fletcher/base.py

```diff
@@ -432,50 +427,20 @@ def __getitem__(self, item):
         For a boolean mask, return an instance of ``ExtensionArray``, filtered
         to the values where ``item`` is True.
         """
-        # Workaround for Arrow bug that segfaults on empty slice.
-        # This is fixed in Arrow master, will be released in 0.10
-        if isinstance(item, slice):
-            start = item.start or 0
-            stop = item.stop if item.stop is not None else len(self.data)
-            stop = min(stop, len(self.data))
-            step = item.step if item.step is not None else 1
-            # Arrow can't handle slices with steps other than 1
-            # https://issues.apache.org/jira/browse/ARROW-2714
-            if step != 1:
-                arr = np.asarray(self)[item]
-                # ARROW-2806: Inconsistent handling of np.nan requires adding a mask
-                if pa.types.is_integer(self.dtype.arrow_dtype) or pa.types.is_floating(
-                    self.dtype.arrow_dtype
-                ):
-                    mask = pd.isna(arr)
-                else:
-                    mask = None
-                return type(self)(pa.array(arr, type=self.dtype.arrow_dtype, mask=mask))
-            if stop - start == 0:
-                return type(self)(pa.array([], type=self.data.type))
-        elif isinstance(item, Iterable):
-            if not is_array_like(item):
-                item = np.array(item)
-            if is_integer_dtype(item):
-                return self.take(item)
-            elif is_bool_dtype(item):
-                indices = np.array(item)
-                indices = np.argwhere(indices).flatten()
-                return self.take(indices)
+        if is_integer(item):
+            return self.data[item].as_py()
+        if (
```

Comment:
What about putting this after `isinstance(item, slice)` and keep using `is_bool_dtype`?